src/v/transform/tests/transform_processor_test.cc

```diff
@@ -105,6 +109,34 @@ class ProcessorTestFixture : public ::testing::Test {
     uint64_t _error_count = 0;
     probe _probe;
 };
+
+MATCHER_P(SameBatchRef, ref, "") {
+    const model::record_batch& batch = ref.get();
+    const auto& a = arg.copy_records();
+    const auto& b = batch.copy_records();
+    if (a.size() != b.size()) {
+        *result_listener << "expected same size: " << a.size() << " vs "
+                         << b.size();
```

Comment:
For my own edification - `arg` and `result_listener` are standard names? or are they specified somewhere?

Reply:
They are standard names for this macro, see http://google.github.io/googletest/reference/matchers.html#defining-matchers

Reply:
`arg` is a terrible choice...very hard to grep for in the browser 🤷. anyway, gtk, thanks.